BredaUniversityGames/Y2024-25-PR-BB
# Feature/steam input

### Description

- Added Steam Input
- Added Input Actions for both Steam and SDL devices
- Reworked how Input Devices are handled
- Added Steam Input configuration files (which should be officially uploaded to Steamworks later on)
- Added Actions based on Pawel's controller scheme
- Converted camera movement to use actions

Includes both Steam Input and Input Actions, which should have been split up.... But without Input Actions, Steam Input wouldn't be implemented well, as it only works with Input Actions. So it's quite a big PR, here is an overview how the structure looks like:

![InputHnadling drawio](https://github.com/user-attachments/assets/b422af0f-d715-4d9d-8e5c-c20117c440f0)
Digital actions -> Actions involving button inputs
Analog actions -> Action involving axis inputs

### Issues

[Steam input codecks card](https://bubonic-brotherhood.codecks.io/decks/30-controls/card/18q-steam-input)
[Input Actions codecks card](https://bubonic-brotherhood.codecks.io/decks/30-controls/card/1fr-input-actions)

### Test criteria

With Steam (Steam Input Action will be used):
1. Make sure Steam is open in the background
2. Start our program with a controller connected (this is the tricky part as I don't know if all controllers will work as I only could set up the generic controller, each controller type can have their own configuration file file action inputs for Steam)
3. Fly around the scene with the axis inputs, it should work as previously

Without Steam (SDL Input Action will be used):
1. Start our program with a controller connected
2. Fly around the scene with the axis inputs, it should work as previously
